### ISSUE TITLE
Class3D, Empty MultidimArray

### ISSUE THREAD

**Issue by wimhagen** [3dem/relion#541]
Commit 9d75256 stops on error in Class3D, our previous 7f8d1d6 works just fine with the exact same data and settings.

```
=============================
 Expectation iteration 1 of 25 (with 6000 particles)
000/??? sec ~~(,_,">                                                          [oo] Size(Y,X): 24x13 i=[0..23] j=[0..12]
 Empty MultidimArray!
 Size(Y,X): 24x13 i=[0..23] j=[0..12]
 Empty MultidimArray!
-------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------
```

```
------------------------------------------------------------------------
No OpenFabrics connection schemes reported that they were able to be
used on a specific port.  As such, the openib BTL (OpenFabrics
support) will be disabled for this port.

  Local host:           gpu9
  Local device:         mlx5_0
  Local port:           1
  CPCs attempted:       rdmacm, udcm
--------------------------------------------------------------------------
[gpu9.cluster.embl.de:102447] 2 more processes have sent help message help-mpi-btl-openib-cpc-base.txt / no cpcs for port
[gpu9.cluster.embl.de:102447] Set MCA parameter "orte_base_help_aggregate" to 0 to see all help / error messages
 WARNING: The reference pixel size is 0.834892 A, but the smallest pixel size in the data is 0.417446 A!
 WARNING: The reference pixel size is 0.834892 A, but the smallest pixel size in the data is 0.417446 A!
 WARNING: The reference pixel size is 0.834892 A, but the smallest pixel size in the data is 0.417446 A!
in: /tmp/eb-build/RELION_core/3.1.0-beta-20191112_05_9d75256_a/foss-2018b-cu100/relion-9d75256d6f053b408a0c060dabb49ddbca631e89/src/multidim_array.h, line 2912
ERROR: 
Array_by_array: different shapes (+)
terminate called after throwing an instance of 'RelionError'
[gpu9:102466] *** Process received signal ***
[gpu9:102466] Signal: Aborted (6)
[gpu9:102466] Signal code:  (-6)
[gpu9:102466] [ 0] /lib64/libpthread.so.0(+0xf5d0)[0x7fdf9dd645d0]
[gpu9:102466] [ 1] /lib64/libc.so.6(gsignal+0x37)[0x7fdf9d303207]
[gpu9:102466] [ 2] /lib64/libc.so.6(abort+0x148)[0x7fdf9d3048f8]
[gpu9:102466] [ 3] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libstdc++.so.6(_ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x125)[0x7fdf9da58e95]
[gpu9:102466] [ 4] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libstdc++.so.6(+0xa2c86)[0x7fdf9da56c86]
[gpu9:102466] [ 5] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libstdc++.so.6(+0xa1c49)[0x7fdf9da55c49]
[gpu9:102466] [ 6] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libstdc++.so.6(__gxx_personality_v0+0x2e8)[0x7fdf9da565b8]
[gpu9:102466] [ 7] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libgcc_s.so.1(+0x10403)[0x7fdf9d6aa403]
[gpu9:102466] [ 8] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libgcc_s.so.1(_Unwind_RaiseException+0x2b0)[0x7fdf9d6aa910]
[gpu9:102466] [ 9] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libstdc++.so.6(__cxa_throw+0x36)[0x7fdf9da56f06]
[gpu9:102466] [10] relion_refine_mpi_default(_ZN16ObservationModel18getAverageMtfImageEi+0x21a)[0x4b4e3a]
[gpu9:102466] [11] in: /tmp/eb-build/RELION_core/3.1.0-beta-20191112_05_9d75256_a/foss-2018b-cu100/relion-9d75256d6f053b408a0c060dabb49ddbca631e89/src/multidim_array.h, line 2912
ERROR: 
Array_by_array: different shapes (+)
relion_refine_mpi_default(_ZN16ObservationModel11divideByMtfEiR13MultidimArrayI8tComplexIdEEbb+0x9b)[0x4b588b]
[gpu9:102466] [12] terminate called after throwing an instance of 'RelionError'
relion_refine_mpi_default(_Z27getFourierTransformsAndCtfsI15MlOptimiserCudaEvlR21OptimisationParamtersR18SamplingParametersP11MlOptimiserPT_13AccPtrFactoryi+0x4e80)[0x6c5e40]
[gpu9:102466] [13] relion_refine_mpi_default(_Z27accDoExpectationOneParticleI15MlOptimiserCudaEvPT_mi13AccPtrFactory+0x29d)[0x6dabdd]
[gpu9:102466] [14] [gpu9:102461] *** Process received signal ***
relion_refine_mpi_default(_ZN15MlOptimiserCuda32doThreadExpectationSomeParticlesEi+0xda)[0x69edea]
[gpu9:102466] [15] [gpu9:102461] Signal: Aborted (6)
[gpu9:102461] Signal code:  (-6)
relion_refine_mpi_default(_Z36globalThreadExpectationSomeParticlesR14ThreadArgument+0x2c)[0x6527dc]
[gpu9:102466] [16] [gpu9:102461] [ 0] relion_refine_mpi_default(_Z11_threadMainPv+0x36)[0x496886]
[gpu9:102466] [17] /lib64/libpthread.so.0(+0xf5d0)[0x7f9a6c05f5d0]
[gpu9:102461] [ 1] /lib64/libpthread.so.0(+0x7dd5)[0x7fdf9dd5cdd5]
[gpu9:102466] [18] /lib64/libc.so.6(gsignal+0x37)[0x7f9a6b5e7207]
[gpu9:102461] [ 2] /lib64/libc.so.6(clone+0x6d)[0x7fdf9d3caead]
[gpu9:102466] *** End of error message ***
/lib64/libc.so.6(abort+0x148)[0x7f9a6b5e88f8]
[gpu9:102461] [ 3] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libstdc++.so.6(_ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x125)[0x7f9a6bd53e95]
[gpu9:102461] [ 4] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libstdc++.so.6(+0xa2c86)[0x7f9a6bd51c86]
[gpu9:102461] [ 5] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libstdc++.so.6(+0xa1c49)[0x7f9a6bd50c49]
[gpu9:102461] [ 6] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libstdc++.so.6(__gxx_personality_v0+0x2e8)[0x7f9a6bd515b8]
[gpu9:102461] [ 7] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libgcc_s.so.1(+0x10403)[0x7f9a7707b403]
[gpu9:102461] [ 8] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libgcc_s.so.1(_Unwind_RaiseException+0x2b0)[0x7f9a7707b910]
[gpu9:102461] [ 9] /g/easybuild/x86_64/CentOS/7/skylake/software/GCCcore/7.3.0/lib64/libstdc++.so.6(__cxa_throw+0x36)[0x7f9a6bd51f06]
[gpu9:102461] [10] relion_refine_mpi_default(_ZN16ObservationModel18getAverageMtfImageEi+0x21a)[0x4b4e3a]
[gpu9:102461] [11] relion_refine_mpi_default(_ZN16ObservationModel11divideByMtfEiR13MultidimArrayI8tComplexIdEEbb+0x9b)[0x4b588b]
[gpu9:102461] [12] relion_refine_mpi_default(_Z27getFourierTransformsAndCtfsI15MlOptimiserCudaEvlR21OptimisationParamtersR18SamplingParametersP11MlOptimiserPT_13AccPtrFactoryi+0x4e80)[0x6c5e40]
[gpu9:102461] [13] relion_refine_mpi_default(_Z27accDoExpectationOneParticleI15MlOptimiserCudaEvPT_mi13AccPtrFactory+0x29d)[0x6dabdd]
[gpu9:102461] [14] relion_refine_mpi_default(_ZN15MlOptimiserCuda32doThreadExpectationSomeParticlesEi+0xda)[0x69edea]
[gpu9:102461] [15] relion_refine_mpi_default(_Z36globalThreadExpectationSomeParticlesR14ThreadArgument+0x2c)[0x6527dc]
[gpu9:102461] [16] relion_refine_mpi_default(_Z11_threadMainPv+0x36)[0x496886]
[gpu9:102461] [17] /lib64/libpthread.so.0(+0x7dd5)[0x7f9a6c057dd5]
[gpu9:102461] [18] /lib64/libc.so.6(clone+0x6d)[0x7f9a6b6aeead]
[gpu9:102461] *** End of error message ***
/g/easybuild/x86_64/CentOS/7/skylake/software/RELION_core/3.1.0-beta-20191112_05_9d75256_a-foss-2018b-cu100/bin/relion_refine_mpi: line 8: 102466 Aborted                 $TOOLNAME$TOOLSUFFIX $@
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[56097,1],2]
  Exit code:    134
--------------------------------------------------------------------------
Command exited with non-zero status 134
	Command being timed: "mpirun -n 3 /g/easybuild/x86_64/CentOS/7/skylake/software/RELION_core/3.1.0-beta-20191112_05_9d75256_a-foss-2018b-cu100/bin/relion_refine_mpi --o Class3D/job014/run --i Extract/job012/particles.star --ref bin1_20A.mrc --ini_high 20 --dont_combine_weights_via_disc --scratch_dir /tmp/46620777 --pool 30 --pad 2 --ctf --ctf_corrected_ref --iter 25 --tau2_fudge 4 --particle_diameter 178 --fast_subsets --K 4 --flatten_solvent --zero_mask --solvent_mask MaskCreate/job004/mask.mrc --oversampling 1 --healpix_order 2 --offset_range 10 --offset_step 4 --sym O --norm --scale --j 3 --gpu  --pipeline_control Class3D/job014/"
	User time (seconds): 776.23
	System time (seconds): 17.23
	Percent of CPU this job got: 99%
	Elapsed (wall clock) time (h:mm:ss or m:ss): 13:18.73
	Average shared text size (kbytes): 0
	Average unshared data size (kbytes): 0
	Average stack size (kbytes): 0
	Average total size (kbytes): 0
	Maximum resident set size (kbytes): 5592756
	Average resident set size (kbytes): 0
	Major (requiring I/O) page faults: 9
	Minor (reclaiming a frame) page faults: 4480518
	Voluntary context switches: 10000
	Involuntary context switches: 51034
	Swaps: 0
	File system inputs: 639928
	File system outputs: 94
	Socket messages sent: 0
	Socket messages received: 0
	Signals delivered: 0
	Page size (bytes): 4096
	Exit status: 134
```

**Comment by biochem-fan** [comment]
Can you try commits in between?

**Comment by biochem-fan** [comment]
This is probably caused by 72574ef.

**Comment by frozenfas** [comment]
I had a similar error continuing a Refine3D that ran fine under 7f8d1d but failed with 885dff.  If I switch to 0841d0 it runs again. I mean I just reverted to 0841d0 and it is currently running and it is past the point it was failing in 885dff.

**Comment by wimhagen** [comment]
> Can you try commits in between?

Unfortunately not as we went from 7f8d1d6 to 9d75256.

**Comment by wimhagen** [comment]
Same problem in 3D auto-refine.

**Comment by biochem-fan** [comment]
Hi, @scheres made a fix. I am testing it right now.

**Comment by biochem-fan** [comment]
Fixed in 62c5915. Please try again and re-open this issue if the problem persists.
